pkg/broker/binding_operations.go

```diff
@@ -139,20 +141,36 @@ func userFromParams(bindingID string, password string, rawParams []byte) (*atlas
 		}
 	}
 
-	// Set binding ID as username and add password.
-	params.User.Username = bindingID
-	params.User.Password = password
-
-	// If no role is specified we default to read/write on any database.
-	// This is the default role when creating a user through the Atlas UI.
-	if len(params.User.Roles) == 0 {
-		params.User.Roles = []atlas.Role{
-			atlas.Role{
-				Name:         "readWriteAnyDatabase",
-				DatabaseName: "admin",
-			},
+	// If the plan ID is specified we construct the provider object from the service and plan.
+	// The plan ID is required for binding
+	if planID != "" {
+		provider, err := b.findProviderByServiceID(serviceID)
```

Comment:
Add a comment here explaining why we are checking the service ID and plan ID when they are never used (the specification requires us to).

Reply:
Yes, I will :)